### PR TITLE
dependable@2.0.0

### PR DESCRIPTION
**Pull request by davemo**
This rewrite removes the following features:
- `container.load`: Load uses synchronous APIs from Node's `fs` module. If users want to load files, 
we feel they should do so externally and then register the dependency.
- `container.register(hash)`: Hash registration adds too much intelligence in the library. If users really want to do this, they can register multiple dependencies using `.map` or a loop.

This eases the burden of complexity on the library. We also cleaned up tests and dropped coffeescript.

Would be great to get some review on this from @Schoonology, @searls, @rosston.

Todo once merged: 
- `npm version major`
- draft a release on github

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,147 +1,63 @@
...
-### Register a dependency that has other dependencies
-
-When the argument is a function, the function's arguments are automatically
-populated with the correct dependencies, and the return value of the function
-is registered as the dependency:
-
-```js
-container.register('song', function (occupation, transport, legalStatus) {
-  var song = {};
+To register a new dependency, call `container.register`:
```
Comment (Schoonology) [inline README.md]: This can go in another PR, but I think we should go into a _lot_ more detail around what a "dependency" is: dependable only cares about the _return value_ of this function, etc. etc. etc.

Comment (primarilysnark) [reply README.md]: Agreed. The main point of the README rewrite was to get rid of a lot of verbiage about writing a song in favor of a more concise example. The README as a whole needs some work.

**Comment by jasonkarns** [comment]
`git push --follow-tags` can replace `git push && git push --tags`

`"postversion": "git push --follow-tags && npm publish"` :D

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        subject.register("goats", function(cans) {
+          return cans;
+        });
+        assert.throws(() => subject.get("goats"));
+      });
+    });
+
+    context("internals", function() {
+      it('should register the container as "_container"', function() {
+        assert.equal(subject.get("_container"), subject);
```
Comment (Schoonology) [inline test/test.js]: What is the use case for this feature?

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  describe("#register", function() {
+    it("should register a factory", function() {
+      subject.register("logger", function() {
+        return "message";
+      });
+      assert.equal(subject.get("logger"), "message");
+    });
+
+    it("should register an object", function() {
+      subject.register("logger", "message");
```
Comment (Schoonology) [inline test/test.js]: I'm on the fence about this feature. Especially in ES6, the difference in code seems trivial (e.g. `() => "message"`). That would require being able to parse arrow functions, though.

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,298 @@
+const container = require("../index.js").container;
+const assert = require("assert");
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
```
Comment (Schoonology) [inline test/test.js]: What are these last three `require` statements for?

Comment (davemo) [reply test/test.js]: Dead code! They were only used for the `.load` feature, good catch 💃

**Comment by jasonkarns** [comment]
Worth adding standard and prettier standard to catch that stuff?

File: test/test.js (hunk 2 above)
Comment (davemo) [reply test/test.js]: Not sure, I believe the original intent was to expose internals if you wanted to do some sort of meta programming around the guts of the container?

File: test/test.js (hunk 3 above)
Comment (davemo) [reply test/test.js]: Well, we had also batted around the idea of a new api similar to what angular 1 did, `container.constant(name, string)`. So maybe it's better to codify this functionality that way? I do sort of agree with what you're saying tho.

File: test/test.js (hunk 2 above)
Comment (Schoonology) [reply test/test.js]: I vote for dropping this until a clear use case presents itself. Even then, if metaprogramming is that desirable, I'd rather an explicit "plugin" API.

File: test/test.js (hunk 3 above)
Comment (primarilysnark) [reply test/test.js]: Depending on the general consensus of `.constant`, I have a working version with tests ready to go.

Comment (Schoonology) [reply test/test.js]: For this first version, I'd rather focus on making the API _smaller_ and better-tested. Once we have confidence in the core, we can branch out into different improvements. (For example, can we get away from function parsing? How do we make it easier to register _functions_ as dependencies?)

Comment (davemo) [reply test/test.js]: Well, we could pull in esprima and do some AST parsing to see if that yields an alternative approach, but then we're a bit heavier on the dependency side.

Comment (davemo) [reply test/test.js]: Also, for additional features we had talked about the idea of a plugin architecture, perhaps `dependable-test`, and `dependable-ext` for additional features? That way we can grow features without adding bloat to the core, well-tested, library.

File: index.js [hunk 5]
```diff
@@ -98,7 +98,5 @@ exports.container = () => {
     list: () => factories
   };
 
-  container.register("_container", container);
```
Comment (davemo) [inline index.js]: 👍

File: LICENSE [hunk 6]
```diff
@@ -1,5 +1,7 @@
 Copyright (c) 2013 i.TV LLC
```
Comment (Schoonology) [inline LICENSE]: At this point I'd say it's safe to drop i.TV from the list.

File: README.md [hunk 7]
```diff
@@ -1,160 +1,71 @@
...
+```JavaScript
+container.constant('config', {
+  password: '********'
 });
 ```
 
-### Override dependencies at resolve time
-
-It's also possible to override dependencies at resolve time:
+To get a registered dependency, simply use `dependable.get`:
```
Comment (davemo) [inline README.md]: `container.get`?

Comment (primarilysnark) [reply README.md]: Good catch!

**Review by Schoonology — APPROVED** [review]
@primarilysnark @davemo This is beautiful. Let's ship it.
